src/utp_wrapper.rs

```diff
@@ -47,7 +49,13 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    if timeout < ::time::SteadyTime::now() {
```

Comment:
The timeout **always** happen, because we've set a timeout [here](https://github.com/maidsafe/crust/pull/442/files#diff-68671ba860f232f621493a8fac59d30fR38). This timeout purpose is to allow crust to break out of read cycle and check if there are pending messages to be sent.

What your code is doing is "drop the connection it the crust user don't schedule any messages to deliver within this amount of time".